tests/test_run_config_generator.py

```diff
@@ -0,0 +1,184 @@
+            profile_models:
+                - my-model0
+                - my-model1
+            """)
+        # yapf: enable
+
+        expected_pa_configs = len(
+            utils.generate_log2_list(DEFAULT_RUN_CONFIG_MAX_CONCURRENCY))
+        expected_model_configs = DEFAULT_RUN_CONFIG_MAX_INSTANCE_COUNT + 1
+        expected_num_of_configs = expected_pa_configs * expected_model_configs**2
+        self._run_and_test_run_config_generator(
+            yaml_content, expected_config_count=expected_num_of_configs)
+
+    def test_default_config_four_models(self):
+        """
+        Test Default Four Models:
+            - No CLI options specified
+
+        log2(DEFAULT_RUN_CONFIG_MAX_CONCURRENCY)+1 PA configs x
+        DEFAULT_RUN_CONFIG_MAX_INSTANCE_COUNT+1 Model configs
+        will be generated by the auto-search
+        """
+
+        # yapf: disable
+        yaml_content = convert_to_bytes("""
+            profile_models:
+                - my-model0
+                - my-model1
+                - my-model2
+                - my-model3
+            """)
+        # yapf: enable
+
+        expected_pa_configs = len(
+            utils.generate_log2_list(DEFAULT_RUN_CONFIG_MAX_CONCURRENCY))
+        expected_model_configs = DEFAULT_RUN_CONFIG_MAX_INSTANCE_COUNT + 1
+        expected_num_of_configs = expected_pa_configs * expected_model_configs**4
+
+        self._run_and_test_run_config_generator(
+            yaml_content, expected_config_count=expected_num_of_configs)
+
+    def _run_and_test_run_config_generator(self,
+                                           yaml_content,
+                                           expected_config_count,
+                                           pa_cli_args=None):
+        args = [
+            'model-analyzer', 'profile', '--model-repository', 'cli_repository',
+            '-f', 'path-to-config-file'
+        ]
+
+        if type(pa_cli_args) == list:
+            args = args + pa_cli_args
+        elif type(pa_cli_args) == str:
+            args.append(pa_cli_args)
+
+        protobuf = """
+            max_batch_size: 8
+            instance_group [
+            {
+                kind: KIND_CPU
+                count: 1
+            }
+            ]
+            """
+
+        self.mock_model_config = MockModelConfig(protobuf)
+        self.mock_model_config.start()
+        config = self._evaluate_config(args, yaml_content)
+
+        rcg = RunConfigGenerator(config, config.profile_models, MagicMock())
+
+        run_configs = []
+        rcg_config = rcg.next_config()
+        while not rcg.is_done():
+            run_configs.append(next(rcg_config))
+
+        self.assertEqual(expected_config_count, len(set(run_configs)))
+
+        # Checks that each RunConfig contains the expected number of model_configs
+        model_configs = [
+            run_config.model_configs() for run_config in run_configs
+        ]
+
+        (self.assertEqual(len(config.profile_models),
```

Comment:
can this statement be rewritten/expanded? it's hard to read the parentheses and understand what's going on

Reply:
I've made it as clean as I can. There's no way I know of to reduce this short of not doing list comprehension, which in my opinion would make it less readable

Reply:
interesting, i was thinking for loops written out would be more readable--i'll just resolve this and move on 👍